src/main/java/StringManipulation.java

```diff
@@ -1,33 +1,133 @@
+
+import java.util.Arrays;
+
 public class StringManipulation implements StringManipulationInterface {
 
+    private String stringToManipulate;
+
     @Override
     public String getString() {
-        return null;
+        return stringToManipulate;
     }
 
     @Override
     public void setString(String string) {
+        stringToManipulate = string;
     }
 
     @Override
     public int count() {
-        return 0;
+
+        // If the current stringToManipulate is null, throws an exception
+        if(stringToManipulate == null){
+            throw new NullPointerException("The string is null.");
+        }
+        // If the stringToManipulate is an empty string, returns 0
+        if(stringToManipulate.length()==0){
+            return 0;
+        }
+
+        // split the string by whitespace(s), period, comma, and other symbols
+        // and stores each word into a String array
+        String[] words = stringToManipulate.split("[\\s.(),:?!]+");
+
+        // returns the length of array as the word count
+        return words.length;
```

Comment:
Great work on the count() method!

I appreciate the clear and concise logic you've implemented. The conditional statements effectively handle scenarios where the stringToManipulate is null or empty, ensuring the method behaves correctly and avoids any potential issues.

I also commend your use of regular expressions in splitting the string into words. It's a clever approach that allows for flexibility in handling different symbols and punctuation marks. Regex is very complicated, yet very useful.